Notebooks/transcript_dataframe_builder.ipynb

```diff
@@ -0,0 +1,1264 @@
+    "#from cdp_backend.pipeline.transcript_model import Transcript\n",
+    "\n",
+    "## For working with the data and our local file system\n",
+    "import pandas as pd\n",
+    "import os "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Connect to Council's Data Base"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## For Seattle\n",
+    "fireo.connection(client=Client(\n",
+    "    project=\"cdp-seattle-staging-dbengvtn\",\n",
+    "    credentials=AnonymousCredentials()\n",
+    "))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Connect to Council's File Store"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## For Seattle\n",
+    "fs = GCSFileSystem(project=\"cdp-seattle-staging-dbengvtn\", token=\"anon\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## Fetch details for all available transcripts for council of interest\n",
+    "transcript_model = list(db_models.Transcript.collection.fetch())[0:]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "58"
+      ]
+     },
+     "execution_count": 14,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "## Total transcripts for this particular council at this time\n",
+    "len(transcript_model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Dowload all transcripts to local machine\n",
+    "__Note:__ _The function below is for downloading these transcripts to your local machine and therefore may be time and space intensive._"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## Finish documenting function\n",
+    "\n",
+    "## Function to Download all transcripts locally\n",
+    "def download_transcripts(list_of_transcripts_db_ref, council):\n",
+    "    '''\n",
+    "    Takes a list of database objects, and a chosen council as arguments.\n",
+    "    council argument must be in string format.\n",
+    "    Creates a file for transcripts to download to.\n",
+    "    Downloads database objects to local machine as .json type.\n",
+    "    Transcript files are saved with name incrementally counting up from 0.\n",
+    "    '''\n",
+    "    num = 0  ## For adding index to file names\n",
+    "    for transcripts in range(0, len(list_of_transcripts_db_ref)):\n",
+    "        s = str(num)\n",
+    "        fs.get(transcript_model[0].file_ref.get().uri, \n",
+    "               \"transcripts/\"+council+\"/\"+council+\"-transcript\"+s+\".json\")\n",
```

Comment:
nit: you should use `fstrings` and pathlib instead of hard coded strings + your operating systems `/` path separator. (Windows and Mac use different path seps)

I would make this:
```python
from pathlib import Path

transcripts_dir = Path("transcripts")
council_dir = transcripts_dir / council  # note the division character _not_ "/", pathlib overloads divison to insert the appropriate os.sep for the user
council_dir.mkdir(parents=True, exist_ok=True)
this_transcript_path = council_dir / f"{council}-transcript-{s}.json"

```

Reply:
Also, we have multiple transcripts for meetings, this should select the highest confidence transcript for each meeting too. But again, I think that is just a larger me task to write a general transcript / dataset download function.

Reply:
Honestly didn't even think about seps. I mean doesn't the world revolve around me?

I was pushing this particular notebook especially for you to look at.

Reply:
I have text cleaned/processed and some EDA with transcripts but I was curious if the way I got there made any sense to someone with experience.